handlers/handlers.go

```diff
@@ -121,3 +223,73 @@ func getRequestBody(request *http.Request) []byte {
 
 	return body
 }
+
+// Token Util
+const oneWeekInMinutes = 60 * 24 * 7
+const cerealNotesCookieName = "CerealNotesToken"
+
+func parseTokenFromString(tokenAsString string) (*jwt.Token, error) {
+	return jwt.ParseWithClaims(
+		strings.TrimSpace(tokenAsString),
+		&JwtTokenClaim{},
+		func(*jwt.Token) (interface{}, error) {
+			return tokenSigningKey, nil
+		})
+}
+
+func createTokenAsString(
+	userId models.UserId,
+	expirationTimeInMinutes int64,
+) (string, error) {
+	claims := JwtTokenClaim{
+		userId,
+		jwt.StandardClaims{
+			ExpiresAt: time.Now().Unix() + (expirationTimeInMinutes * 60),
+			Issuer:    "CerealNotes",
+		},
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	return token.SignedString(tokenSigningKey)
+}
+
+func getUserIdFromJwtToken(request *http.Request) (models.UserId, error) {
+	cookie, err := request.Cookie(cerealNotesCookieName)
+	if err != nil {
+		return -1, err
+	}
+
+	token, err := parseTokenFromString(cookie.Value)
+	if err != nil {
+		return -1, err
+	}
+
+	if claims, ok := token.Claims.(*JwtTokenClaim); ok && token.Valid {
+		return claims.userId, nil
+	}
+	return -1, errors.Errorf("Token was invalid or unreadable")
+}
+
+func tokenTest1() {
```

Comment:
reminder to move to another file

Reply:
previous comment says I'll do it after you create testing framework